elide-graphql/src/main/java/com/yahoo/elide/graphql/parser/GraphQLEntityProjectionMaker.java

```diff
@@ -349,11 +354,10 @@ private void addArgument(Argument argument, EntityProjectionBuilder projectionBu
             addSorting(argument, projectionBuilder);
         } else if (ModelBuilder.ARGUMENT_FILTER.equals(argumentName)) {
             addFilter(argument, projectionBuilder);
-        } else if (isEntityArgument(argumentName, entityDictionary, projectionBuilder.getType())) {
-            addEntityArgument(argument, projectionBuilder);
         } else if (!ModelBuilder.ARGUMENT_OPERATION.equals(argumentName)
                 && !(ModelBuilder.ARGUMENT_IDS.equals(argumentName))
-                && !(ModelBuilder.ARGUMENT_DATA.equals(argumentName))) {
+                && !(ModelBuilder.ARGUMENT_DATA.equals(argumentName))
+                && !isEntityArgument(argumentName, entityDictionary, projectionBuilder.getType())) {
```

Comment:
It still gets invoked for calls like this `author(id: "1")`  It will add an argument in projection for `id`. also checks if someone is passing an argument that is not valid like `author(unknown: "1")`. `unknown` is not a valid attribute.